packages/test/local-server-stress-tests/src/stressDataObject.ts

```diff
@@ -0,0 +1,320 @@
+		const channels: IChannel[] = [];
+		for (const [name] of this.channelNameMap.entries()) {
+			// similar to container objects, the entries in this map
+			// can appear before the underlying channel is attached,
+			// so getting the channel can fail, and we need to try
+			// to get all channel each time, as we have no way to
+			// observer when a channel moves from detached to attached,
+			// especially on remove clients/
+			const channel = await this.runtime.getChannel(name).catch(() => undefined);
+			if (channel !== undefined) {
+				channels.push(channel);
+			}
+		}
+		return channels;
+	}
+
+	protected async hasInitialized(): Promise<void> {
+		this.defaultStressObject = await this.getDefaultStressDataObject();
+
+		this.channelNameMap = (await this.runtime.getChannel(
+			"channelNameMap",
+		)) as any as ISharedMap;
+	}
+
+	public get attached() {
+		return this.runtime.attachState === AttachState.Attached;
+	}
+
+	public uploadBlob(tag: `blob-${number}`, contents: string) {
+		void this.runtime.uploadBlob(stringToBuffer(contents, "utf-8")).then((handle) =>
+			this.defaultStressObject.registerLocallyCreatedObject({
+				type: "newBlob",
+				handle,
+				tag,
+			}),
+		);
+	}
+
+	public createChannel(tag: `channel-${number}`, type: string) {
+		this.runtime.createChannel(tag, type);
+		this.channelNameMap.set(tag, type);
+	}
+
+	public createDataStore(tag: `datastore-${number}`, asChild: boolean) {
+		void this.context.containerRuntime
+			.createDataStore(
+				asChild
+					? [...this.context.packagePath, StressDataObject.factory.value.type]
+					: StressDataObject.factory.value.type,
+			)
+			.then(async (dataStore) => {
+				const maybe: FluidObject<StressDataObject> | undefined =
+					await dataStore.entryPoint.get();
+				assert(maybe?.StressDataObject !== undefined, "must be stressDataObject");
+				this.defaultStressObject.registerLocallyCreatedObject({
+					type: "stressDataObject",
+					handle: dataStore.entryPoint,
+					tag,
+					stressDataObject: maybe.StressDataObject,
+				});
+			});
+	}
+}
+export type ContainerObjects =
+	| { type: "newBlob"; handle: IFluidHandle; tag: `blob-${number}` }
+	| {
+			type: "stressDataObject";
+			tag: `datastore-${number}`;
+			handle: IFluidHandle;
+			stressDataObject: StressDataObject;
+	  };
+
+export class DefaultStressDataObject extends StressDataObject {
+	public static readonly alias = "default";
+
+	public get DefaultStressDataObject() {
+		return this;
+	}
+
+	/**
+	 * these are object created in memory by this instance of the datastore, they
+	 * will also be in  these the containerObjectMap, but are not necessarily usable
+	 * as they could be detached, in which can only this instance can access them.
+	 */
+	private readonly _locallyCreatedObjects: ContainerObjects[] = [];
+	public async getContainerObjects(): Promise<readonly Readonly<ContainerObjects>[]> {
+		const globalObjects: Readonly<ContainerObjects>[] = [...this._locallyCreatedObjects];
```

Comment:
Naming is a little confusing here since the local objects are not global, maybe consider localAndGlobalObjects or allObjects